File: shortest-word-1.js
// Time, Space - O(N), O(1)

var shortestDistance = function(words, word1, word2) {
    let j = -1;
    let k = -1;
    let min = Number.MAX_VALUE;
    
    for(let i=0;i<words.length;i++) {
        if(word1 === words[i]) {
            j = i;
        }
        if(word2 === words[i]) {
            k = i;
        }
        if(j!==-1 && k!==-1) {
            min = Math.min(Math.abs(j-k), min);   
        }
    }
    
    return min;
};

File: shortestWord1.java
// Time Complexity :O(n)
// Space Complexity :constant
// Did this code successfully run on Leetcode :yes
// Any problem you faced while coding this :no
class Solution {
    public int shortestDistance(String[] wordsDict, String word1, String word2) {
        // we'll use two pointers to find minimum distance
        int p1 = -1;
        int p2 = -1;
        int min = Integer.MAX_VALUE;
        for (int i = 0; i < wordsDict.length; i++) {
            String curr = wordsDict[i];
            if (curr.equals(word1)) {
                p1 = i;
            } else if (curr.equals(word2)) {
                p2 = i;
            }
            // take minimum distance if we found both words
            if (p1 != -1 && p2 != -1) {
                min = Math.min(min, Math.abs(p1 - p2));
            }
        }
        return min;
    }
}
